lib/protocol/protocol.js

```diff
@@ -418,22 +419,14 @@ function driverShouldDoJwpProxy (driver, req, command) {
 
   // validate avoidance schema, and say we shouldn't proxy if anything in the
   // avoid list matches our req
-  let proxyAvoidList = driver.getProxyAvoidList(req.params.sessionId);
-  for (let avoidSchema of proxyAvoidList) {
-    if (!_.isArray(avoidSchema) || avoidSchema.length !== 2) {
-      throw new Error('Proxy avoidance must be a list of pairs');
-    }
-    let [avoidMethod, avoidPathRegex] = avoidSchema;
-    if (!_.includes(['GET', 'POST', 'DELETE'], avoidMethod)) {
-      throw new Error(`Unrecognized proxy avoidance method '${avoidMethod}'`);
-    }
-    if (!(avoidPathRegex instanceof RegExp)) {
-      throw new Error('Proxy avoidance path must be a regular expression');
-    }
-    let normalizedUrl = req.originalUrl.replace(/^\/wd\/hub/, '');
-    if (avoidMethod === req.method && avoidPathRegex.test(normalizedUrl)) {
-      return false;
-    }
+  if (driver.proxyRouteIsAvoided(req.params.sessionId, req.method, req.originalUrl)) {
+    return false;
+  }
+
+  // if it looks like we have an image element in the body, never proxy
+  const stringBody = JSON.stringify(req.body);
+  if (stringBody && stringBody.includes(IMAGE_ELEMENT_PREFIX)) {
```

Comment:
wouldn't it be safer to change includes to `/`\b${IMAGE_ELEMENT_PREFIX}`/.test(stringBody)` ?